### PR TITLE
Close stream

### PR DESCRIPTION
**Pull request by hadrienk**
This PR add tests that ensure the streams (`getData()`) used in the operations are closed.

The UnfoldOperation and CrossJoinOperation were not closing the streams. This was fixed by upgrading to protonpack 1.15 and ensuring no `orElse()` is used on the `Optional<Stream<DataPoint>>` result of `getData()`.

### CODE REVIEW

File: java-vtl-script/src/test/java/no/ssb/vtl/script/operations/JoinAssignmentTest.java [hunk 1]
```diff
@@ -147,9 +152,53 @@ public void testImplicitFails() throws Exception {
...
+
+    @Test
+    public void testNewMeasureValue() {
+
+        JoinAssignment operation;
+        operation = new JoinAssignment(
+                dataset,
+                expression,
+                "measure",
+                Component.Role.ATTRIBUTE,
```
Comment (takvamborgen) [inline java-vtl-script/src/test/java/no/ssb/vtl/script/operations/JoinAssignmentTest.java]: Component.Role.MEASURE?

File: java-vtl-script/src/test/java/no/ssb/vtl/script/support/DatasetCloseWatcher.java [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+    private void increase() {
+        this.counter++;
+    }
+
+    private void decrease() {
+        this.counter--;
+    }
+
+    private Stream<DataPoint> wrap(Stream<DataPoint> stream) {
+        this.increase();
```
Comment (pawbu) [inline java-vtl-script/src/test/java/no/ssb/vtl/script/support/DatasetCloseWatcher.java]: I'm wondering if increasing and decreasing the counter should be thread-safe or not.

File: java-vtl-script/src/test/java/no/ssb/vtl/script/operations/FilterOperationTest.java [hunk 3]
```diff
@@ -20,23 +20,21 @@
...
 import javax.script.Bindings;
 
+import java.util.stream.Stream;
+
 import static org.assertj.core.api.Assertions.assertThat;
 
 public class FilterOperationTest {
 
-    private Dataset dataset;
+    private DatasetCloseWatcher dataset;
```
Comment (pawbu) [inline java-vtl-script/src/test/java/no/ssb/vtl/script/operations/FilterOperationTest.java]: Nice idea with the watcher, but it leaves me wondering if this variable here is a dataset or a watcher. Maybe have the new methods under Dataset interface?

File: java-vtl-script/src/test/java/no/ssb/vtl/script/support/DatasetCloseWatcher.java [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+    @Override
+    public Optional<Stream<DataPoint>> getData(Filtering filtering) {
+        return getData(filtering).map(this::wrap);
+    }
+
+    @Override
+    public Optional<Stream<DataPoint>> getData(Set<String> components) {
+        return getData(components).map(this::wrap);
+    }
+
```
Comment (takvamborgen) [inline java-vtl-script/src/test/java/no/ssb/vtl/script/support/DatasetCloseWatcher.java]: This method seems to recurse infinitely.

File: java-vtl-script/src/test/java/no/ssb/vtl/script/support/DatasetCloseWatcher.java [hunk 5]
```diff
@@ -0,0 +1,90 @@
...
+    @Override
+    public Optional<Stream<DataPoint>> getData(Order order) {
+        return delegate().getData(order).map(this::wrap);
+    }
+
+    @Override
+    public Optional<Stream<DataPoint>> getData(Filtering filtering) {
+        return getData(filtering).map(this::wrap);
+    }
+
```
Comment (takvamborgen) [inline java-vtl-script/src/test/java/no/ssb/vtl/script/support/DatasetCloseWatcher.java]: This method seems to recurse infinitely.

File: java-vtl-script/src/test/java/no/ssb/vtl/script/operations/JoinAssignmentTest.java (hunk 1 above)
Comment (hadrienk) [reply java-vtl-script/src/test/java/no/ssb/vtl/script/operations/JoinAssignmentTest.java]: Yeah maybe something isn't right here

File: java-vtl-script/src/test/java/no/ssb/vtl/script/operations/FilterOperationTest.java (hunk 3 above)
Comment (hadrienk) [reply java-vtl-script/src/test/java/no/ssb/vtl/script/operations/FilterOperationTest.java]: Yeah I see your point, this only for testing though. If we add this check inside the script package we could think of a better place.

File: java-vtl-script/src/test/java/no/ssb/vtl/script/support/DatasetCloseWatcher.java (hunk 2 above)
Comment (hadrienk) [reply java-vtl-script/src/test/java/no/ssb/vtl/script/support/DatasetCloseWatcher.java]: Thought of it, I don't think it is necessary.

**Pull request merged**